rust/basic_bitcoin/src/basic_bitcoin/src/bitcoin_api.rs

```diff
@@ -0,0 +1,91 @@
+use ic_btc_types::{
+    GetBalanceRequest, GetCurrentFeePercentilesRequest, GetUtxosRequest, GetUtxosResponse,
+    MillisatoshiPerByte, Network, SendTransactionRequest,
+};
+use ic_cdk::{api::call::call_with_payment, export::Principal};
+
+// The fees for the various bitcoin endpoints.
+const GET_BALANCE_COST_CYCLES: u64 = 100_000_000;
+const GET_UTXOS_COST_CYCLES: u64 = 100_000_000;
+const GET_CURRENT_FEE_PERCENTILES_FEE: u64 = 100_000_000;
+const SEND_TRANSACTION_BASE_FEE: u64 = 5_000_000_000;
+const SEND_TRANSACTION_FEE_PER_BYTE: u64 = 20_000_000;
+
+/// Returns the balance of the given bitcoin address.
+///
+/// Relies on the `bitcoin_get_balance` endpoint.
+/// See https://internetcomputer.org/docs/current/references/ic-interface-spec/#ic-bitcoin_get_balance
+pub async fn get_balance(network: Network, address: String) -> u64 {
+    let balance_res: Result<(u64,), _> = call_with_payment(
+        Principal::management_canister(),
+        "bitcoin_get_balance",
+        (GetBalanceRequest {
+            address,
+            network,
+            min_confirmations: None,
+        },),
+        GET_BALANCE_COST_CYCLES,
+    )
+    .await;
+
+    balance_res.unwrap().0
+}
+
+/// Returns the UTXOs of the given bitcoin address.
+///
+/// Relies on the `bitcoin_get_utxos` endpoint.
+/// See https://internetcomputer.org/docs/current/references/ic-interface-spec/#ic-bitcoin_get_utxos
+pub async fn get_utxos(network: Network, address: String) -> GetUtxosResponse {
+    let utxos_res: Result<(GetUtxosResponse,), _> = call_with_payment(
+        Principal::management_canister(),
+        "bitcoin_get_utxos",
+        (GetUtxosRequest {
+            address,
+            network,
+            filter: None,
+        },),
+        GET_UTXOS_COST_CYCLES,
+    )
+    .await;
+
+    utxos_res.unwrap().0
```

Comment:
We should note in the documentation that pagination is ignored here.

Reply:
Agreed, done.